applications/updater/views/updater_main.c

```diff
@@ -0,0 +1,150 @@
+#include <gui/gui_i.h>
+#include <gui/view.h>
+#include <gui/elements.h>
+#include <gui/canvas.h>
+#include <furi.h>
+#include <input/input.h>
+
+#include "../updater_i.h"
+#include "updater_main.h"
+
+struct UpdaterMainView {
+    View* view;
+    ViewDispatcher* view_dispatcher;
+    FuriPubSubSubscription* subscription;
+    void* context;
+};
+
+static const uint8_t PROGRESS_RENDER_STEP = 3; /* percent, to limit rendering rate */
+
+typedef struct {
+    string_t status;
+    uint8_t progress, rendered_progress;
+    uint8_t idx_stage, total_stages;
+    bool failed;
+} UpdaterProgressModel;
+
+void updater_main_model_set_state(
+    UpdaterMainView* main_view,
+    const char* message,
+    uint8_t progress,
+    uint8_t idx_stage,
+    uint8_t total_stages,
+    bool failed) {
+    with_view_model(
+        main_view->view, (UpdaterProgressModel * model) {
+            model->failed = failed;
+            model->idx_stage = idx_stage;
+            model->total_stages = total_stages;
+            model->progress = progress;
+            if(string_cmp_str(model->status, message)) {
+                model->rendered_progress = 101; /* to force view update */
```

Comment:
Looks strange. You can just `return true`.

Reply:
Done